packages/components/src/lib/MarkdownViewer.tsx

```diff
@@ -40,53 +58,114 @@ export const MarkdownViewer: React.FC<MarkdownViewerProps> = ({
   backgroundColor = "bg-slate-50",
 }) => {
   return (
-    <ReactMarkdown
-      className={clsx(
-        "prose",
-        className,
-        textColor || "prose-stone",
-        textSize === "sm" ? "text-sm" : "text-xs"
-      )}
-      rehypePlugins={[rehypeInlineCodeProperty]}
-      remarkPlugins={[remarkGfm]}
-      components={{
-        pre({ children }) {
-          return (
-            <pre
-              className={clsx(
-                "not-prose my-1 rounded p-2 text-[.9em]",
-                backgroundColor
-              )}
-            >
-              {children}
-            </pre>
-          );
-        },
-        code(props) {
-          const { node, children, className, ...rest } = props;
-          const match = /language-(\w+)/.exec(className || "");
-          const isInline = node && node.properties["inline"];
-          if (isInline === "true") {
+    <>
+      <style>{codeBlockStyles}</style>
+      <ReactMarkdown
+        className={clsx(
+          "prose",
+          className,
+          textColor || "prose-stone",
+          textSize === "sm" ? "text-sm" : "text-xs"
+        )}
+        rehypePlugins={[rehypeInlineCodeProperty, rehypeRaw]}
+        remarkPlugins={[remarkGfm]}
+        components={{
+          pre: ({ children }) => <React.Fragment>{children}</React.Fragment>,
+          code({ node, className, children, ...rest }) {
+            const match = /language-(\w+)/.exec(className || "");
+            const isInline =
+              node && (node as Element).properties["inline"] === "true";
+
+            if (isInline) {
+              return (
+                <code
+                  {...rest}
+                  className="not-prose break-words rounded-sm border border-black border-opacity-[0.04] bg-black bg-opacity-[0.03] px-1 py-0.5 font-mono text-[.9em]"
+                >
+                  {children}
+                </code>
+              );
+            }
+
+            // Handle nested code blocks
+            if (
+              match &&
+              match[1] === "typescript" &&
+              String(children).includes("```")
+            ) {
+              return (
+                <div
+                  className={clsx(
+                    "code-block-wrapper overflow-hidden rounded",
+                    backgroundColor
+                  )}
+                >
+                  <div className="p-4">
+                    <pre>
+                      <code className="language-typescript">
+                        {String(children)}
+                      </code>
+                    </pre>
+                  </div>
+                </div>
+              );
+            }
+
+            if (match && match[1] === "squigglePlayground") {
```

Comment:
squigglePlayground is a bit broken now (doesn't properly expand), but I think this does make us a bit of progress.